src/Microsoft.Extensions.Logging/Logger.cs

```diff
@@ -9,59 +9,19 @@ namespace Microsoft.Extensions.Logging
 {
     internal class Logger : ILogger
     {
-        private readonly LoggerFactory _loggerFactory;
-        private readonly string _categoryName;
-        private LoggerInformation[] _loggers;
-        private readonly Func<string, LogLevel, bool> _categoryFilter;
-
-        public Logger(LoggerFactory loggerFactory, string categoryName, Func<string, LogLevel, bool> categoryFilter)
-        {
-            _loggerFactory = loggerFactory;
-            _categoryName = categoryName;
-
-            var providers = loggerFactory.GetProviders();
-            if (providers.Length > 0)
-            {
-                _loggers = new LoggerInformation[providers.Length];
-                for (var index = 0; index < providers.Length; index++)
-                {
-                    _loggers[index] = new LoggerInformation
-                    {
-                        Logger = providers[index].Key.CreateLogger(categoryName),
-                        // Order of preference
-                        // 1. Custom Name
-                        // 2. Provider FullName
-                        ProviderNames = new List<string>
-                        {
-                            providers[index].Value,
-                            providers[index].Key.GetType().FullName
-                        }
-                    };
-                }
-            }
-
-            _categoryFilter = categoryFilter;
-        }
+        public LoggerInformation[] Loggers { get; set; }
 
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
-            if (_loggers == null)
+            if (Loggers == null)
             {
                 return;
             }
 
             List<Exception> exceptions = null;
-            foreach (var loggerInfo in _loggers)
+            foreach (var loggerInfo in Loggers)
```

Comment:
Race condition; `Loggers` is public and might have been set to null between this and the check above.